### ISSUE TITLE
Back function from event details should lead back to favorite list when started from there

### ISSUE THREAD

**Issue by bjalbor** [EventFahrplan/EventFahrplan#703]
# Acknowledgments
Please check the following boxes with an `x` if they apply:
* [x] I have read the latest version of the [contribution guide](
  https://github.com/EventFahrplan/EventFahrplan/blob/master/CONTRIBUTING.md) before creating this pull request.
* [x] I have checked a few other issues labeled `Feature request` to see how they are written.
* [x] The feature I want to propose would be useful for the majority of users, not only for me personally.
* [x] I am aware that _EventFahrplan_ is mostly developed by one person in their unpaid spare time.
* [ ] I can help myself to get this feature implemented or know someone who wants to do it.

# Preliminaries
When using the App to find the perfect personal timetable, it would bei nice if you could make the best choice beteeen overlapping sessions by quickly comparing the event details.

# Current state
After changing into event details from favorites the back function leads to main timetable.

# Target state
After changing into event details from favorites the back function should lead to back to favorite list.

# Hints

**Comment by johnjohndoe** [comment]
@bjalbor Thank you for your feedback.

There are two ways to navigate back: 
1. Use the arrow button in the top right. It will navigate you to the main screen
2. Use the system back button or swipe gesture. It will take you to the list of favorites.

Does this resolve the issue for you?

**Comment by bjalbor** [comment]
@johnjohndoe Didn't notice the difference beteeen both back buttons, thanks for pointing out. 

But then IMO to be clear either the symbol of the arrow button top right should be a home symbol like a house. Or it should take the same action like the other one.

**Comment by johnjohndoe** [comment]
I had a quick read into the [Android documentation](https://developer.android.com/guide/navigation/principles) and it might be that they changed the concept of the "Up" button over the years. I will re-evaluate the concept.

**Comment by tessarakt** [comment]
Same behaviour exists for search results.

But now at least I know that I need to use the system back button.

@johnjohndoe If you make any changes based on your re-evaluation of the concept, I believe that should behave the same. Probably there is more similar navigation logic in the app ...

**Comment by johnjohndoe** [comment]
@tessarakt Yes, exactly. The **back arrow** (aka. home/up button) takes you **out of the navigation hierarchy** to the schedule screen ("home"). This is what Google defined in the past as [stated earlier](https://github.com/EventFahrplan/EventFahrplan/issues/703#issuecomment-2564441691).

**Comment by cbeyls** [comment]
This is a common report. I think most people are not aware of the difference between the back and up buttons. Google is to be blamed.
Another side effect of mistakenly using the up button as back button is that, even if you land back on the same screen with both, there is no guarantee that the state will be preserved when using the up button, so for example you'll always end up at the top of a list instead of preserving its scroll position.

I still consider the hierarchical up navigation as a useful added feature so I keep it in my apps but many people disagree and believe up should behave the same as back. The debate is still on.
